app/src/main/java/ch/epfl/sweng/studdybuddy/core/Buddy.java

```diff
@@ -0,0 +1,55 @@
+package ch.epfl.sweng.studdybuddy.core;
+
+import java.util.Date;
+
+public final class Buddy {
+    public String getAlice() {
+        return alice;
+    }
+
+    public void setAlice(String alice) {
+        this.alice = alice;
+    }
+
+    public String getBob() {
+        return bob;
+    }
+
+    public void setBob(String bob) {
+        this.bob = bob;
+    }
+
+    public Long getCreationDate() {
+        return creationDate;
+    }
+
+    public void setCreationDate(Long creationDate) {
+        this.creationDate = creationDate;
+    }
+
+    private String alice;
+    private String bob;
+    private Long creationDate;
+    public Buddy(){}
+    public Buddy(String alice, String bob) {
+        creationDate = new Date().getTime();
+        this.alice = alice;
+        this.bob = bob;
+    }
+
+    //Returns null if not buddy, because no Optionnal for Java<8
+    public String buddyOf(String uid) {
+        String buddy = null;
+        if(alice.equals(uid)) {
+            buddy = bob;
+        }
+        else if(bob.equals(uid)) {
+            buddy = alice;
+        }
+        return buddy;
+    }
+
+    public String hash() {
+        return Integer.toHexString(alice.hashCode() ^ bob.hashCode());
```

Comment:
Note: This hash function will not translate well to real world deployment. It is "not clear" if symmetric, collision resistant hash function is even achievable, might need to give up on symmetricity.